Sort headers alphabetically in the Network tab of Web Inspector.

eepskamp opened the pull request:
#### 2a225e5c1bcfa6149d0fcde8f7871e93e910e974
<pre>
Sort headers alphabetically in the Network tab of Web Inspector.
<a href="https://bugs.webkit.org/show_bug.cgi?id=243569">https://bugs.webkit.org/show_bug.cgi?id=243569</a>

Reviewed by Timothy Hatcher and Devin Rousso.

* Source/WebInspectorUI/UserInterface/Views/ResourceHeadersContentView.js:
(WI.ResourceHeadersContentView.prototype._createSortedArrayForHeaders):
Create helper method for sorting the headers.
(WI.ResourceHeadersContentView.prototype._refreshRedirectHeadersSections):
(WI.ResourceHeadersContentView.prototype._refreshRequestHeadersSection):
(WI.ResourceHeadersContentView.prototype._refreshResponseHeadersSection):
Deploy helper method.

Canonical link: <a href="https://commits.webkit.org/253138@main">https://commits.webkit.org/253138@main</a>
</pre>

dcrousso commented on Source/WebInspectorUI/UserInterface/Views/ResourceHeadersContentView.js:
```diff
@@ -376,9 +381,7 @@ WI.ResourceHeadersContentView = class ResourceHeadersContentView extends WI.Cont
-        let requestHeaders = this._resource.requestHeaders;
-        for (let key in requestHeaders)
-            this._requestHeadersSection.appendKeyValuePair(key, requestHeaders[key], "header");
+        this._requestHeadersSection.appendKeyValuePair(":path", WI.h2Path(urlComponents), "h2-pseudo-header");
```
oops i think you meant this
```js
for (let [key, value] of this._createSortedArrayForHeaders(this._resource.requestHeaders))
    this._requestHeadersSection.appendKeyValuePair(key, value, "header");
```

eepskamp replied:
Great catch! That is what I meant.

Commit:
Sort headers alphabetically in the Network tab of Web Inspector.
https://bugs.webkit.org/show_bug.cgi?id=243569

Reviewed by Timothy Hatcher and Devin Rousso.

* Source/WebInspectorUI/UserInterface/Views/ResourceHeadersContentView.js:
(WI.ResourceHeadersContentView.prototype._createSortedArrayForHeaders):
Create helper method for sorting the headers.
(WI.ResourceHeadersContentView.prototype._refreshRedirectHeadersSections):
(WI.ResourceHeadersContentView.prototype._refreshRequestHeadersSection):
(WI.ResourceHeadersContentView.prototype._refreshResponseHeadersSection):
Deploy helper method.

Canonical link: https://commits.webkit.org/253138@main
